collectors/VMPropertiesCollector.py

```diff
@@ -55,11 +57,17 @@ def do_metrics(self, target, g, i):
                         continue
                     data = value_entry['data']
                     vm_id = value_entry['resourceId']
+                    project_id = "internal"
+                    if project_ids:
+                        for pid in project_ids:
+                            if vm_id in pid.keys():
```

Comment:
`.keys()` is not needed

Reply:
It works that way. So I wouldn't change it.

Reply:
As long as you don't need the watch of the dynamic changes in the dict object there is no reason to use it. 
(Ref: https://docs.python.org/3.1/library/stdtypes.html#dict.keys)